src/main/kotlin/com/mdrobnak/lalrpop/inspections/MissingTerminalsInspection.kt

```diff
@@ -0,0 +1,57 @@
+package com.mdrobnak.lalrpop.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.LocalInspectionToolSession
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiElementVisitor
+import com.mdrobnak.lalrpop.psi.*
+
+object MissingTerminalsInspection : LocalInspectionTool() {
+    var unresolved = mutableListOf<PsiElement>()
+    var terminalDefs = mutableListOf<PsiElement>()
```

Comment:
The visitor returned from `buildVisitor` needs to be thread-safe, but it currently modifies these lists, which are not thread-safe. These lists should be changed to a thread-safe version (maybe use `Collections.synchronizedList`?), or the algorithm should be changed to avoid mutable state (might not be feasible/worthwhile).